### PR TITLE
[`flake8-pyi`] Add `PYI032` rule with autofix

### PR DESCRIPTION
**Pull request by qdegraaf**
## Summary

Adds Y032 rule from `flake8-pyi` (https://github.com/PyCQA/flake8-pyi) to the Ruff plugin. Original rule definition: 

> The second argument of an __eq__ or __ne__ method should usually be annotated with object rather than Any.

## Test Plan

Fixtures for `.py` and `.pyi` added and added both cases to existing `flake8-pyi` unit tests 

## Issue link

Refers: https://github.com/charliermarsh/ruff/issues/848

### CODE REVIEW

File: crates/ruff/src/rules/flake8_pyi/rules/any_eq_ne_annotation.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            {
+                let mut diagnostic = Diagnostic::new(
+                    AnyEqNeAnnotation {
+                        method_name: name.to_string(),
+                    },
+                    args.args[1].range,
+                );
+                if checker.patch(diagnostic.kind.rule()) {
+                    // def __eq__(self, arg2: Any): ...
+                    if let Some(name) = annotation.as_name_expr() {
```
Comment (qdegraaf) [inline crates/ruff/src/rules/flake8_pyi/rules/any_eq_ne_annotation.rs:76]: I tried combining this logic with the the checker logic from lines 64-66 in a more efficient and elegant manner. Did not succeed in finding a better way, though I believe there are many ways to Rome here, so welcome any advice on making this patch more compact.

Comment (charliermarsh) [reply crates/ruff/src/rules/flake8_pyi/rules/any_eq_ne_annotation.rs]: Hmm -- is there any issue with just using the range of the annotation? Why do we need to match on the "kind"?

**Comment by charliermarsh** [comment]
Thanks!

File: crates/ruff/src/rules/flake8_pyi/rules/any_eq_ne_annotation.rs (hunk 1 above)
Comment (qdegraaf) [reply crates/ruff/src/rules/flake8_pyi/rules/any_eq_ne_annotation.rs]: Ah I see what I missed now. Tried that first, but could not get it to work. But I forgot to import: `use ruff_python_ast::prelude::Ranged;`

Thanks for edits!

**Pull request merged**